src/Pact/Typechecker.hs

```diff
@@ -539,8 +545,8 @@ unifyTypes l r = case (l,r) of
         (TypeVar {},TyVar SchemaVar {}) -> Nothing
         (TypeVar {},TyUser {}) -> Nothing
         (TypeVar _ ac,TyVar sv@(TypeVar _ bc)) -> case unifyConstraints ac bc of
-          Just (Left uc) -> Just $ vWrap $ TyVar $ v { _tvConstraint = uc }
-          Just (Right uc) -> Just $ sWrap $ TyVar $ sv { _tvConstraint = uc }
+          Just (Left uc)  -> Just . vWrap . TyVar $ (v  & tvConstraint .~ uc)
+          Just (Right uc) -> Just . sWrap . TyVar $ (sv & tvConstraint .~ uc)
```

Comment:
Revert this style-only change, and consult the lens style guide.

Reply:
This one isn't strictly style - the use of `Traversal'` here fixes a non-exhaustive pattern match warning. 

There's quite a bit of lens use around that doesn't follow the new Lens guide, but I can alter this usage here to match.

Reply:
Yes, analyze code in particular predates the guide, and is for all intents and purposes a sub-project so is best left to B&J to arbitrate. Outside of Pact.Analyze, the guide should be reflected. Here is a good example of why operators are ungainly, `set tvConstraint uc sv` is much clearer, and doesn't require switching to the backward evaluation direction.

Reply:
But yes, code was unsafe so this is a good change overall.

Reply:
Oh wow, yeah, `set ...` is much cleaner. That's refreshing - I've been using the operators for so long that using the named alternatives didn't even occur to me. But, the operators shall see :fire: !